### PR TITLE
fix: allow accounts to be nullable in `readContract` and add `sim_getTransactionsForAddress` method

### PR DESCRIPTION
**Pull request by AgustinRamiroDiaz**
# What

- fixed typo
- used constant instead of hardcoded string
- allowed accounts to be nullable in `readContract`
- added `sim_getTransactionsForAddress` method

# Why

- continuous improvements
- needed the features / fixes

# Testing done

Added unit tests

# Decisions made

# Checks

- [x] I have tested this code
- [x] I have reviewed my own PR
- [ ] I have created an issue for this PR
- [x] I have set a descriptive PR title compliant with [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)

# Reviewing tips

# User facing release notes

### CODE REVIEW

**Comment by AgustinRamiroDiaz** [comment]
@cristiam86 @denishacquin  do you know how to impose typescript checks in the tests? I'm being able to just test logic, but not typing

**Comment by cristiam86** [comment]
> @cristiam86 @denishacquin do you know how to impose typescript checks in the tests? I'm being able to just test logic, but not typing

I think that typing should be static, not tested. Do you have a concrete example?

File: tests/client.test.ts [hunk 1]
```diff
@@ -9,3 +12,102 @@ describe("Client Creation", () => {
...
+          from: overrideAccount.address,
+          data: expect.any(String),
+        },
+        "latest",
+      ],
+    });
+  });
+});
+
+describe("Request Method", () => {
```
Comment (cristiam86) [inline tests/client.test.ts:81]: @AgustinRamiroDiaz it seems that the expects are missing here

**Review by cristiam86 — CHANGES_REQUESTED** [review]

File: tests/client.test.ts (hunk 1 above)
Comment (AgustinRamiroDiaz) [reply tests/client.test.ts:81]: @cristiam86 these are examples of what I wanted to ask. In these tests, I only want to test the typing of the `request` method. How can I test that? If I put any garbage in it, the test still passes

**Comment by AgustinRamiroDiaz** [comment]
I've also fixed all eslint errors so test pass

**Review by cristiam86 — APPROVED** [review]

**Pull request merged**